packages/jest-message-util/src/index.ts

```diff
@@ -90,6 +90,29 @@ const getRenderedCallsite = (
   return renderedCallsite;
 };
 
+function checkForCommonEnvironmentErrors(error: string) {
+  if (
+    error.includes('ReferenceError: document is not defined') ||
+    error.includes('ReferenceError: window is not defined')
+  ) {
+    return warnAboutWrongTestEnvironment(error, 'jsdom');
+  } else if (error.includes('.unref is not a function')) {
+    return warnAboutWrongTestEnvironment(error, 'node');
+  }
+
+  return error;
+}
+
+function warnAboutWrongTestEnvironment(error: string, env: 'jsdom' | 'node') {
+  return (
+    chalk.red(
+      chalk.bold(
+        `Jest: You should use a ${env} test environment, see https://jestjs.io/docs/en/configuration#testenvironment-string.\n\n`,
```

Comment:
Suggestion:
```
The error below may be caused by using the wrong test environment, see <link>.
Consider using the <env> test environment.
```

Reply:
I was about to suggest the same :). BTW, I think it would be nice to show the link with and underline and maybe a dim, like: `chalk.underline.dim(link)`

Reply:
Thanks for the suggestions!

![image](https://user-images.githubusercontent.com/1404810/55277742-bea4f600-5303-11e9-9434-521b45ba93be.png)

(`dim` doesn't work in my theme, for some reason)